File: catamesh-core/src/main/java/dev/catamesh/application/facade/DefaultTemplateFacade.java
package dev.catamesh.application.facade;

import dev.catamesh.core.cqrs.Query;
import dev.catamesh.core.facade.TemplateFacade;

public class DefaultTemplateFacade implements TemplateFacade {

    private final Query<String, String> getFileFromResourceQuery;

    public DefaultTemplateFacade(Query<String, String> getFileFromResourceQuery) {
        this.getFileFromResourceQuery = getFileFromResourceQuery;
    }

    @Override
    public String initDataProduct(String dataProductName) {
        String yaml = getFileFromResourceQuery.execute("templates/data-product.v1.template.yaml");
        return yaml.replace("${NAME}", dataProductName);
    }
}

File: catamesh-cli/src/application/facade/DefaultTemplateFacade.ts
import {TemplateFacade} from "../../core/facade/TemplateFacade";
import {Query} from "../../core/cqrs/Query";
import {CataMeshCoreCommand} from "../../infrastructure/cqrs/CataMeshCoreCommand";
import * as fs from "node:fs";

export class DefaultTemplateFacade implements TemplateFacade {

    private cataMeshCoreCommand: Query<string[], string> = new CataMeshCoreCommand();

    newTemplate(command: string[]): void {
        //todo validate input
        const yaml: string = this.cataMeshCoreCommand.execute(command);
        fs.writeFileSync(command[2] + ".yaml", yaml, "utf8");
    }
}